File: 백준/Silver/1463. 1로 만들기/1로 만들기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {


    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
//        StringBuilder sb = new StringBuilder();
//        StringTokenizer st = new StringTokenizer(br.readLine(), " ");
        int N = Integer.parseInt(br.readLine());
        System.out.println(recur(N, 0));

    }

    static int recur(int N, int count) {
        if (N < 2) return count;
        return Math.min(recur(N / 2, count + 1 + (N % 2)), recur(N / 3, count + 1 + (N % 3)));
    }
}

File: 백준/Silver/1463. 1로 만들기/1로 만들기.py
X = int(input())
count = 0
def recur(X, count):
    if X < 2 :
        return count
    return min(recur(X//2 , count + 1 + (X % 2)), recur(X//3 , count + 1 + (X % 3)))

print(recur(X, count))